src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.OpenFileTracker.cs

```diff
@@ -215,30 +247,51 @@ void WatchHierarchy(IVsHierarchy hierarchyToWatch)
                 }
 
                 // We may have multiple projects with the same hierarchy, but we can use __VSHPROPID8.VSHPROPID_ActiveIntellisenseProjectContext to distinguish
+                VisualStudioProject? project = null;
                 if (ErrorHandler.Succeeded(hierarchy.GetProperty(VSConstants.VSITEMID_ROOT, (int)__VSHPROPID8.VSHPROPID_ActiveIntellisenseProjectContext, out var contextProjectNameObject)))
```

Comment:
when would this fail?

Reply:
This is only supported for multi-targeting projects, so it's only supported from CPS. Other project systems (csproj, the legacy web projects, third party ones) won't.